src/bytes.rs

```diff
@@ -400,10 +400,6 @@ impl Bytes {
             len,
         );
 
-        if end == begin {
-            return Bytes::new();
-        }
```

Comment:
Right now, the `Bytes` instance is going to hold a refcount to the allocation, which is not necessary. Please go through `new_empty_with_ptr` instead.

Reply:
I see, this is a very nifty technique! That means that in principle, you could have `b1: Bytes` at address `a`, take an empty slice via `let s1 = b1.slice(..0)`, then have `b1` destroyed. Next, you could have `b2: Bytes ` at the *same* address `a`, and take another empty slice via `let s2 = b2.slice(..0)`. The two slices `s1` and `s2` came from different Bytes, yet share the same address. I thought that this would be a problem, but it isn't, because `s1` and `s2`  are empty anyway.
Cool stuff!